std/datetime.d

```diff
@@ -2433,28 +2438,65 @@ public:
 
 
     /++
-        Returns a $(D time_t) which represents the same time as this
-        $(LREF SysTime).
+        Converts this $(LREF SysTime) to unix time (i.e. seconds from midnight,
+        January 1st, 1970 in UTC).
 
-        Note that like all conversions in std.datetime, this is a truncating
-        conversion.
+        On most systems, unix time is equivalent to time_t, but that depends on
+        the C implementation as the exact representation of time_t is
+        implementation-defined (e.g. it does not appear that the Digital Mars C
+        runtime uses unix time for time_t; it appears to apply the local DST to
+        time_t).
+
+        By default, the return type is the same number of bits as time_t (which
+        is normally 64 bits on 64-bit systems and 32 bits on 32-bit systems. If
+        64 bits is required on a 32-bit system, then pass $(D 64) as a template
+        argument (or if 32 bits is required on a 64-bit system, pass $(D 32)).
+
+        If the return value is 32 bits, and the result can't fit in a 32-bit
+        value, then the closest value that can be held in 32 bits will be used
+        (so $(D int.max) if it goes over and $(D int.min) if it goes
+        under). However, no attempt is made to deal with integer overflow if
+        the return type is 64 bits.
 
-        If $(D time_t) is 32 bits, rather than 64, and the result can't fit in a
-        32-bit value, then the closest value that can be held in 32 bits will be
-        used (so $(D time_t.max) if it goes over and $(D time_t.min) if it goes
-        under).
+        Params:
+            bits = The number of bits in the return type (32 or 64). It
+                   defaults to 32 on 32-bit systems and 64 on 64-bit systems.
+
+        Returns:
+            A signed integer representing the unix time which is equivalent to
+            this SysTime.
       +/
-    time_t toUnixTime() @safe const pure nothrow
+    auto toUnixTime(ubyte bits = time_t.sizeof * 8)() @safe const pure nothrow
+        if(bits == 32 || bits == 64)
```

Comment:
I suppose so. It didn't occur me. I was thinking 32-bit vs 64-bit and ended up with this, but using `int` and `long` would probably work just as well.